util/merkle/benches/triedb.rs

```diff
@@ -0,0 +1,165 @@
+use std::sync::Arc;
+
+use cmerkle::{Trie, TrieMut, TrieDB, TrieDBMut};
+use kvdb::DBTransaction;
+use primitives::H256;
+use rand::random;
+use rocksdb::{CompactionProfile, Database, DatabaseConfig};
+use tempfile::{tempdir, TempDir};
+use test::Bencher;
+
+struct TestDB {
+    _dir: TempDir,
+    db: Arc<Database>,
+    journal: Box<dyn journaldb::JournalDB>,
+    root: H256,
+}
+
+impl TestDB {
+    // Replicate CodeChain's db config
+    fn config(path: &Path) -> DatabaseConfig {
+        let mut config = DatabaseConfig::with_columns(Some(1));
+        config.memory_budget = Default::default();
+        config.compaction = CompactionProfile::auto(path);
+        config.wal = true;
+        config
+    }
+
+    fn populate(path: &Path, size: usize) -> H256 {
+        // Create database
+        let config = Self::config(path);
+        let db = Arc::new(Database::open(&config, path.to_str().unwrap()).unwrap());
+        let mut journal = journaldb::new(db.clone(), journaldb::Algorithm::Archive, Some(0));
+        let mut root = H256::new();
+        {
+            let hashdb = journal.as_hashdb_mut();
+            let mut trie = TrieDBMut::new(hashdb, &mut root);
+            for i in 0..size {
+                trie.insert(&i.to_be_bytes(), &i.to_be_bytes()).unwrap();
+            }
+        }
+        let mut batch = DBTransaction::new();
+        journal.journal_under(&mut batch, 0, &H256::new());
+        db.write(batch).unwrap();
+
+        root
+    }
+
+    fn new(size: usize) -> Self {
+        // Create temporary directory
+        let dir = tempdir().unwrap();
+        let root = Self::populate(dir.path(), size);
+
+        // Create database
+        let config = Self::config(dir.path());
+        let db = Arc::new(Database::open(&config, dir.path().to_str().unwrap()).unwrap());
+        let journal = journaldb::new(db.clone(), journaldb::Algorithm::Archive, Some(0));
+
+        Self {
+            _dir: dir,
+            db,
+            journal,
+            root,
+        }
+    }
+
+    fn trie(&self) -> TrieDB {
+        let hashdb = self.journal.as_hashdb();
+        TrieDB::try_new(hashdb, &self.root).unwrap()
+    }
+
+    fn trie_mut(&mut self) -> TrieDBMut {
+        let hashdb = self.journal.as_hashdb_mut();
+        TrieDBMut::new(hashdb, &mut self.root)
+    }
+
+    fn flush(&mut self) {
+        let mut batch = DBTransaction::new();
+        self.journal.journal_under(&mut batch, 0, &H256::new()).unwrap();
+        self.db.write(batch).unwrap();
+        self.db.flush().unwrap();
+    }
+}
+
+const DB_SIZE: usize = 10000;
+const BATCH: usize = 10000;
+
+#[bench]
+fn bench_read_single(b: &mut Bencher) {
+    let mut db = TestDB::new(DB_SIZE);
+    b.iter(|| {
+        let trie = db.trie();
+        let item = random::<usize>() % DB_SIZE;
+        let bytes = trie.get(&item.to_be_bytes()).unwrap().unwrap();
+        assert_eq!(item, usize::from_be_bytes((*bytes).try_into().unwrap()));
+        db.flush();
```

Comment:
Any reason to call flush to test reading?

Reply:
No specific reason for that. It was just one of my trial-an-errors.
I think we don't need it though. Do you agree with that?

Reply:
Yes, I think we don't need it to test reading.